app/internal/world_clock.py

```diff
@@ -0,0 +1,512 @@
+        time="Early evening",
+        start="17:00:00",
+        end="18:59:59",
+        desirability="Can be considered",
+    ),
+    MeetingTime(
+        time="Evening",
+        start="19:00:00",
+        end="20:59:59",
+        desirability="Better not",
+    ),
+    MeetingTime(
+        time="Night",
+        start="21:00:00",
+        end="23:59:59",
+        desirability="Better not",
+    ),
+    MeetingTime(
+        time="Late night",
+        start="00:00:00",
+        end="04:59:59",
+        desirability="Not possible",
+    ),
+]
+
+
+def normalize_continent_name(continent_name: str) -> Optional[str]:
+    """Normalize a given continent name to the continent name
+       that appears in the timezone list.
+    Args:
+        continent_name (str): The given continent name.
+    Returns:
+        str: The normalized continent name.
+    """
+    for continent in CONTINENTS:
+        if continent in continent_name:
+            return continent
+    return continent_name
+
+
+def load_country_continent_data_set() -> List[Dict[str, str]]:
+    """Load the country-continent data set.
+
+    Returns:
+        list: The country-continent data set
+    """
+    with open(COUNTRY_TO_CONTINENT_DATA_SET_PATH) as file:
+        return json.load(file)
+
+
+def get_continent(country_name: str) -> Optional[str]:
+    """Get the continent of a given country.
+    Args:
+        country_name (str): The given country name.
+    Returns:
+        str: The suitable continent name.
+    """
+    details = load_country_continent_data_set()
+    for country_element in details:
+        if (country_name.title() in
+                country_element[COUNTRY_NAME_KEY_IN_CONTITNENTS]):
+            return normalize_continent_name(
+                country_element[CONTINENT_NAME_KEY]
+            )
+
+
+def load_city_country_data_set() -> List[Dict[str, str]]:
+    """Load the city-country data set.
+
+    Returns:
+        list: The city-country data set.
+    """
+    with open(CITY_TO_COUNTRY_DATA_SET_PATH) as file:
+        return json.load(file)
+
+
+def get_country(city_name: str) -> Optional[str]:
+    """Get the country of a city.
+    Args:
+        city_name (str): The given city name.
+    Returns:
+        str: The suitable country name.
+    """
+    details = load_city_country_data_set()
+    for city_element in details:
+        if city_name.title() in city_element[CITY_NAME_KEY].title():
+            return city_element[COUNTRY_NAME_KEY_IN_CITIES]
+
+
+def get_subcountry(city_name: str) -> Optional[str]:
+    """Get the subcountry of a city.
+    Args:
+        city_name (str): The given city name.
+    Returns:
+        str: The suitable subcountry name.
+    """
+    details = load_city_country_data_set()
+    for city_element in details:
+        if city_name.title() in city_element[CITY_NAME_KEY].title():
+            return city_element[SUBCOUNTRY_NAME_KEY_IN_CITIES]
+
+
+async def get_api_data(url: str) -> Optional[List[Any]]:
```

Comment:
Cache this function to prevent high loading times

Reply:
I did my best to do so by using the decorator `@functools.lru_cache` but I always get runtime errors with the message: "cannot reuse already awaited coroutine". The truth is I'm pretty desperate and do not know how to solve this. I probably won't have the time to work on the code again by the end of the submission deadline - Can you consider giving up this matter or help me here? I have worked so hard...

Reply:
Have you tried [async-lru](https://github.com/aio-libs/async-lru)?